packages/block-editor/src/store/selectors.js

```diff
@@ -283,19 +296,8 @@ export const getClientIdsOfDescendants = createSelector(
  *
  * @return {Array} ids of top-level and descendant blocks.
  */
-export const getClientIdsWithDescendants = createSelector(
-	( state ) => {
-		const collectedIds = [];
-		for ( const topLevelId of getBlockOrder( state ) ) {
-			collectedIds.push(
-				topLevelId,
-				...getClientIdsOfDescendants( state, topLevelId )
-			);
-		}
-		return collectedIds;
-	},
-	( state ) => [ state.blocks.order ]
-);
+export const getClientIdsWithDescendants = ( state ) =>
```

Comment:
Not related to your changes, but I've always thought it's awkward when writing application code that we have _both_ of these functions. They're named similarly and I can never remember which is which. Could we deprecate `getClientIdsWithDescendants` and make it so that `getClientIdsOfDescendants` defaults to `rootIds = ''`?

Reply:
Yes, it's really weird that we have both of these functions, I can deprecate it in a follow-up